proposed/2021/TargetFrameworkAndPlatformsFilters.md

```diff
@@ -0,0 +1,196 @@
+
+There are many unique version numbers for all generations of .NET & supported platforms. The filtering experience should be smart enough to create a version list based on the current selection. Here are a few scenarios:
+
+- User selects `Any` for the .NET generation.
+  - Version filter populates all known versions for .NET Framework, .NET Core, .NET, and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8, 5.0, 6.0
+- User selects `Any` for the .NET generation and `Windows` for the supported platform.
+  - Version filter populates all known version for .NET Framework, .NET Core, .NET, and .NET Standard.
+  - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8, 5.0, 6.0
+- User selects `Any` for the .NET generation and `Android` for the supported platform.
+  - Version filter populates all known version for .NET and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 6.0
+- User selects `.NET Framework` for the .NET generation.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8
+- User selects `.NET Framework` for the .NET generation and `Windows` for the supported platform.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8
+- User selects `.NET Core` for the .NET generation and `macOS` for the supported platform.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1
+
+For further scenarios, see the [framework precedence](https://nugettools.azurewebsites.net/6.0.0-preview.3/framework-precedence) tool to understand how the version matrix would be calculated.
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/VersionNumbers.png)
+
+#### Complete Experience
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/CompleteExperience.png)
+
+## Search Syntax
+
+Lastly, NuGet should add a protocol in which supports [search syntax](https://docs.microsoft.com/nuget/consume-packages/finding-and-choosing-packages#search-syntax) for frameworks, platforms, and potentially *implementations**.
+
+Here's an ideal `<property>:<term>` for each use-case:
+
+```
+framework:net472 framework:net471 framework:net47   # Match the target framework monikers
+
+platform:windows platform:macos platform:linux    # Match the platforms
+
+implementation:netframework implementation:netcore implementation:net  # Match the generation of .NET including both supported frameworks & platforms.
+```
+
+- `framework:` - Allows a short-form TFM value to search on.
```

Comment:
Thanks Andy. `supportedFramework` will help with the technical explanation of the spec if we choose to re-use it. It does require a fully qualified name such as `.NETFramework,Version=v4.5` vs. the shorthand `<TargetFramework>` that you might include in your project file. We'll have to see how easy it is to support common names & whether the existing functionality will suffice.

Reply:
One special consider woudl be how we handle things such as `net5.0-windows`. 

There is an implicit platform version set by the SDK, that NuGet does not understand. 

This would be a larger concern with net5.0-android. 

Alternatively, we could not respect combinations such as this.

Reply:
What is the concern? 

`net6.0-androidXX.X` would represent `.NET 6` + `Android`. The platform version is not in scope of this spec today if it is defined.

Reply:
Can you mix framework and platform in the queries? 

For example, framework=net5.0-android and platform=macos. 

It's  easy to get into a situation where the queries contradict.

Reply:
Ah gotcha. No you wouldn't be able to do this with modern TFM. It's one framework and multiple platforms.

.NET 6 + Android + macOS = net6.0-android && net6.0-macos

Reply:
So is `net6.0-android` allowed in the framework part?

Reply:
I don't know right now. I would say that if it's a valid short form TFM, then yes. But it should be smart enough to treat the `-<platform>` as a platform filter in the search call. We will see what's possible shortly.

Reply:
> But it should be smart enough to treat the -<platform> as a platform filter in the search call

Yeah, I'd guess that's 100% doable. 
Mixing and matching specific framework and platform would be weird.

Let's see what the customer dev says. 

We can always say you can't mix the early, but then allow it later.

Reply:
Resolving as this has a path forward, adding it under `unresolved questions` with the current answer.